mmv1/products/memorystore/Instance.yaml

```diff
@@ -0,0 +1,341 @@
+    name: zoneDistributionConfig
+    immutable: true
+    default_from_api: true
+    description: "Zone distribution configuration for allocation of instance resources. "
+  - !ruby/object:Api::Type::Boolean
+    name: deletionProtectionEnabled
+    description: "Optional. If set to true deletion of the instance will fail. "
+    default_value: true
+  - !ruby/object:Api::Type::Array
+    name: pscAutoConnections
+    item_type: !ruby/object:Api::Type::NestedObject
+      properties:
+        - !ruby/object:Api::Type::String
+          name: pscConnectionId
+          description:
+            "Output only. The PSC connection id of the forwarding rule connected
+            to the\nservice attachment. "
+          output: true
+        - !ruby/object:Api::Type::String
+          name: ipAddress
+          description:
+            "Output only. The IP allocated on the consumer network for the
+            PSC forwarding rule. "
+          output: true
+        - !ruby/object:Api::Type::String
+          name: forwardingRule
+          description: "Output only. The URI of the consumer side forwarding rule.\nFormat:\nprojects/{project}/regions/{region}/forwardingRules/{forwarding_rule} "
+          output: true
+        - !ruby/object:Api::Type::String
+          name: projectId
+          description:
+            "Output only. The consumer project_id where the forwarding rule is
+            created from. "
+          output: true
+        - !ruby/object:Api::Type::String
+          name: network
+          description:
+            "Output only. The consumer network where the IP address resides, in
+            the form of\nprojects/{project_id}/global/networks/{network_id}. "
+          output: true
+    description:
+      "Output only. User inputs and resource details of the auto-created
+      PSC connections. "
+    output: true
+parameters:
+  - !ruby/object:Api::Type::String
+    name: location
+    description:
+      "Resource ID segment making up resource `name`. It identifies the resource
+      within its parent collection as described in https://google.aip.dev/122. See documentation
+      for resource type `memorystore.googleapis.com/CertificateAuthority`. "
+    url_param_only: true
+    required: true
+    immutable: true
+  - !ruby/object:Api::Type::String
+    name: instanceId
+    description:
+      "Required. The ID to use for the instance, which will become the final
+      component of\nthe instance's resource name.\n\nThis value is subject to the following
+      restrictions:\n\n* Must be 4-63 characters in length\n* Must begin with a letter
+      or digit\n* Must contain only lowercase letters, digits, and hyphens\n* Must not
+      end with a hyphen\n* Must be unique within a location "
+    url_param_only: true
+    required: true
+    immutable: true
+virtual_fields:
+  - !ruby/object:Api::Type::Array
+    name: desired_psc_auto_connections
```

Comment:
After reading how this field is implemented, and taking a quick look at the API requests, I'm wondering why this wasn't implemented as a modifiable field `psc_auto_connections`. Could you briefly explain why this was needed (or point me to other documentation/conversation)? In short, it seems like the API allows `psc_auto_connections` in requests, and returns it in responses as expected.

Reply:
Drive-by comment: yaqs/986109097998811136 has the relevant discussion; TLDR the API returns two entries in psc_auto_connections for every entry _sent_ in psc_auto_connections.

Reply:
Ah thanks, that makes sense then!